debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/snapshot/query/SelectAllSnapshotQuery.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.oracle.snapshot.query;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import io.debezium.bean.StandardBeanNames;
+import io.debezium.bean.spi.BeanRegistry;
+import io.debezium.bean.spi.BeanRegistryAware;
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.connector.oracle.OracleOffsetContext;
+import io.debezium.connector.oracle.OraclePartition;
+import io.debezium.pipeline.spi.Offsets;
+import io.debezium.snapshot.spi.SnapshotQuery;
+
+public class SelectAllSnapshotQuery implements SnapshotQuery, BeanRegistryAware {
+
+    private BeanRegistry beanRegistry;
+
+    @Override
+    public String name() {
+        return CommonConnectorConfig.SnapshotQueryMode.SELECT_ALL.getValue();
+    }
+
+    @Override
+    public void configure(Map<String, ?> properties) {
+
+    }
+
+    @Override
+    public void injectBeanRegistry(BeanRegistry beanRegistry) {
+        this.beanRegistry = beanRegistry;
+    }
+
+    @Override
+    public Optional<String> snapshotQuery(String tableId, List<String> snapshotSelectColumns) {
+
+        final Offsets<OraclePartition, OracleOffsetContext> mySqloffsets = beanRegistry.lookupByName(StandardBeanNames.OFFSETS, Offsets.class);
+        final OracleOffsetContext offset = mySqloffsets.getTheOnlyOffset();
+
+        final String snapshotOffset = offset.getScn().toString();
+        String columns = String.join(", ", snapshotSelectColumns);
+        assert snapshotOffset != null;
+        return Optional.of(String.format("SELECT %s FROM %s AS OF SCN %s", columns, tableId, snapshotOffset));
```

Comment:
If we allow the user to omit the `AS OF SCN %s` bit on a custom implementation, are we comfortable then allowing the user to do an inconsistent data snapshot?  Would that even be something we'd support?

Reply:
Well, it depends on how much flexibility we want to leave. Full or just partially?

Reply:
I'd personally allow that. This is pretty advanced feature so if anybody is going to play with it then we can expect a certain degree of competence.

Reply:
Hi @mfvitale, so `AS OF SCN` basically tells Oracle that the select statement should pull the data that was in the table at that point in time based on the system change number, which effectively allows Oracle to generate a consistent snapshot without any locks.  

If this isn't used, then either the implementation will have to use locks or it won't be consistent.

Reply:
> I'd personally allow that. This is pretty advanced feature so if anybody is going to play with it then we can expect a certain degree of competence.

Understood, that seems fair but we'll probably want to make sure to clarify this point in the documentation around this feature so that users are aware that they'll need to consider that in case they're unaware.

Reply:
> Hi @mfvitale, so `AS OF SCN` basically tells Oracle that the select statement should pull the data that was in the table at that point in time based on the system change number, which effectively allows Oracle to generate a consistent snapshot without any locks.
> 
> If this isn't used, then either the implementation will have to use locks or it won't be consistent.

Thanks for the explanation.